ntbs-service/DataMigration/NotificationImportService.cs

```diff
@@ -41,23 +42,69 @@ public class NotificationImportService : INotificationImportService
             _logger = logger;
         }
 
-        public async Task<IList<Notification>> ImportNotificationsAsync(string requestId, string notificationId)
+
+        public async Task<IList<Notification>> ImportNotificationsByDateAsync(string requestId, DateTime cutoffDate)
         {
-            _logger.LogInformation(requestId, $"Request to import Notification with Id={notificationId}");
+            _logger.LogInformation(requestId, $"Request to import by Date started");
+
+            var notificationsGroups = (await _notificationMapper.GetByDate(cutoffDate)).ToList();
 
-            if (_notificationRepository.NotificationWithLegacyIdExists(notificationId)) 
+            // Filter out notifications that already exist in ntbs database
+            var notificationsGroupsToImport = new List<List<Notification>>();
+            foreach (var notificationsGroup in notificationsGroups)
             {
-                _logger.LogInformation(requestId, $"Notification with Id={notificationId} already exists in database");
-                return null;
+                var notificationsToImport = GetNotificationsToImport(requestId, notificationsGroup.Select(x => x.LTBRID ?? x.ETSID).ToList());
+                if (notificationsToImport.Count() == notificationsGroup.Count())
+                {
+                    notificationsGroupsToImport.Add(notificationsGroup);
+                }

```

Comment:
Let's say the a patient has 3 notifications. This if statement will avoid importing all of it if some of them already exist in database. Do you think it's unnecessary and we will either have all imported or none?

Reply:
I would *hope* that that's the case, but I think we can just assume it will be - so it's probably worth highlighting that fact with a big error. So something like:
`if` (counts match) `then` import `elseif` count is 0 `then` ignore `else` log *error* - unexpected invalid state, manual intervention needed!